### PR TITLE
Feature/product table crud operation

### PR DESCRIPTION
**Pull request by NanditaNikam**

### CODE REVIEW

File: src/modules/user/controllers/product.controller.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+} from '@nestjs/common';
+import { ProductService } from '../services/product.services';
+import { CreateProductDto } from '../dto/create-product.dto';
+import { Product } from '../entities/product.entity';
+
+@Controller('products')
+export class ProductController {
+  constructor(private productService: ProductService) {}
+
+  @Post('create')
```
Comment (ShrisomLaha) [inline src/modules/user/controllers/product.controller.ts:18]: not required any path name

File: src/modules/user/entities/product.entity.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Entity, PrimaryGeneratedColumn, Column, ManyToOne } from 'typeorm';
+import { Category } from './category.entity';
+import { Color } from './color.entity';
+
+@Entity({ name: 'product_master' })
+export class Product {
```
Comment (ShrisomLaha) [inline src/modules/user/entities/product.entity.ts:6]: create one BaseEntity that should contain columns like 

1. id -> primary key
2. created_by
3. updated_by

extend this base entity class to create all entity

File: src/modules/user/services/product.services.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    private colorRepository: Repository<Color>,
+  ) {}
+
+  async createProduct(createProductDto: CreateProductDto): Promise<Product> {
+    const product = new Product();
+    product.name = createProductDto.name;
+    product.amount = createProductDto.amount;
+    product.discount = createProductDto.discount;
+
+    const category = await this.categoryRepository.findOne({
```
Comment (ShrisomLaha) [inline src/modules/user/services/product.services.ts:25]: if we have configured oneto many or many to one relationship in Product entity then we don't need this

**Review by ShrisomLaha — CHANGES_REQUESTED** [review]

File: src/modules/user/controllers/product.controller.ts (hunk 1 above)
Comment (NanditaNikam) [reply src/modules/user/controllers/product.controller.ts:18]: ok will change it

File: src/modules/user/entities/product.entity.ts (hunk 2 above)
Comment (NanditaNikam) [reply src/modules/user/entities/product.entity.ts:6]: ok

File: src/modules/user/services/product.services.ts (hunk 3 above)
Comment (NanditaNikam) [reply src/modules/user/services/product.services.ts:25]: ok

**Pull request merged**